requirements.txt

```diff
@@ -1,85 +1,567 @@
-# NB: Any lines marked with 'append_to_freeze' will be copied over to `requirements-freeze.txt` when creating SCT's
-#     stable releases. We do this because `pip freeze` will not capture options (e.g. --extra-index-url) or
-#     platform-specific requirements (e.g. sys.platform == 'win32')
+#
+# This file is autogenerated by pip-compile with Python 3.9
+# by the following command:
+#
+#    pip-compile spinal_requirements.in
```

Comment:
As far as Windows goes, the error is pretty much the same as before -- `monai` is imported, and the script just [halts abruptly with no error log](https://github.com/spinalcordtoolbox/spinalcordtoolbox/actions/runs/15005195956/job/42179703311?pr=4902#step:6:372):

```
Check if markupsafe (3.0.2) is installed............[OK] (3.0.2)
Check if matplotlib (3.9.4) is installed............[OK] (3.9.4)
Check if matplotlib-inline (0.1.7) is installed.....[OK]
Check if mdurl (0.1.2) is installed.................[OK] (0.1.2)
Check if monai (1.4.0) is installed.................
Error: Process completed with exit code 127.
```

Reply:
Small note about the [other errors](https://github.com/spinalcordtoolbox/spinalcordtoolbox/actions/runs/15005195956/job/42179703311?pr=4902#step:6:43) in the log -- they're not related.

(Our requirements checker doesn't like it when packages have different import names than their package names, so freezing all of the deps like this has caused new errors that are entirely unrelated to `onnx==1.18.0`.)

Reply:
With the test I wanted to see if the dependency resolution works in general and if it's possible to find a combination of all packages.